Foundation/FileManager.swift

```diff
@@ -209,20 +240,93 @@ open class FileManager : NSObject {
                 useDarwinPaths = false
             #endif
         }
-        
-        var urls: [URL] = []
-        
+
         for domain in domains {
             if useDarwinPaths {
                 urls.append(contentsOf: darwinURLs(for: directory, in: domain))
             } else {
                 urls.append(contentsOf: xdgURLs(for: directory, in: domain))
             }
         }
-        
+#endif
+
         return urls
     }
-    #endif
+
+#if os(Windows)
+    private class func getURL(for id: KNOWNFOLDERID) -> URL {
+      var pszPath: PWSTR?
+      _ = withUnsafePointer(to: id) { id in
+        SHGetKnownFolderPath(id, DWORD(KF_FLAG_DEFAULT.rawValue), nil, &pszPath)
+      }
+      let url: URL = URL(fileURLWithPath: String(decodingCString: pszPath!, as: UTF16.self), isDirectory: true)
+      CoTaskMemFree(pszPath)
+      return url
+    }
+
+    private func windowsURLs(for directory: SearchPathDirectory, in domain: _SearchPathDomain) -> [URL] {
+      guard domain == .user else { return [] }
+
+      switch directory {
+      case .autosavedInformationDirectory:
+        // FIXME(compnerd) where should this go?
+        return []
+
+      case .desktopDirectory:
+        return [FileManager.getURL(for: FOLDERID_Desktop)]
+
+      case .documentDirectory:
+        return [FileManager.getURL(for: FOLDERID_Documents)]
+
+      case .cachesDirectory:
+        // FIXME(compnerd) where should this go?
```

Comment:
This should be a temporary directory that is regularly cleaned up by the OS.

Reply:
Hmm, so, this should just return `%TEMP%`?

Reply:
If that's the best we can do, yeah.